src/internal/irmin-generics/dune

```diff
@@ -0,0 +1,5 @@
+(library
+ (name irmin_generics)
+ (public_name irmin-private-libs.irmin-generics)
```

Comment:
why do you need a private library here? Why not just use `public_name irmin.generics` or `irmin.type` :-) and put that in the `irmin` package?

Reply:
My question in response would be "why _not_ use a private library?" :wink: The code in this library has no dependencies on anything else in Irmin, and could be tested / fuzzed independently. Putting it in `irmin.type` makes it accessible to anyone depending on Irmin (and I believe also exposes it in the Irmin documentation).

If you think the lack of proper support for private libraries in Dune is a blocker, I'm happy to use `irmin.type` / `irmin.generics` instead.

Reply:
I don't think we need to add to add the complexity of an extra package right now. As `irmin` is the sole user of that package right now, it's fine to just make it a sub-library and avoiding that extra opam file.

Reply:
OK, will do.